docs/devenv.rst

```diff
@@ -0,0 +1,593 @@
+Mopidy and extensions.
+
+
+Make a virtualenv
+-----------------
+
+Make a Python `virtualenv <https://virtualenv.pypa.io/>`_ for Mopidy
+development. The virtualenv will wall of Mopidy and its dependencies from the
+rest of your system. All development and installation of Python dependencies
+versions of Mopidy and extensions are done inside the virtualenv. This way your
+regular Mopidy install, which you set up in the first step, is unaffected by
+your hacking and will always be working.
+
+Most of us use the `virtualenvwrapper
+<https://virtualenvwrapper.readthedocs.org/>`_ to ease working with
+virtualenvs, so that's what we'll be using for the examples here. First,
+install and setup virtualenvwrapper as described in their docs.
+
+To create a virtualenv named ``mopidy`` which use Python 2.7, allows access to
+system-wide packages like GStreamer, and use the Mopidy workspace directory as
+the "project path", run::
+
+    mkvirtualenv -a ~/mopidy-dev --python `which python2.7` \
+      --system-site-packages mopidy
+
+Now, each time you open a terminal and want to activate the ``mopidy``
+virtualenv, run::
+
+    workon mopidy
+
+This will both activate the ``mopidy`` virtualenv, and change the current
+working directory to ``~/mopidy-dev``.
+
+
+Clone the repo from GitHub
+--------------------------
+
+Once inside the virtualenv, it's time to clone the ``mopidy/mopidy`` Git repo
+from GitHub::
+
+    git clone https://github.com/mopidy/mopidy.git
+
+When you've cloned the ``mopidy`` Git repo, ``cd`` into it::
+
+    cd ~/mopidy-dev/mopidy/
+
+With a fresh clone of the Git repo, you should start out on the ``develop``
+branch. This is where all features for the next feature release lands. To
+confirm that you're on the right branch, run::
+
+    git branch
+
+
+Install development tools
+-------------------------
+
+We use a number of Python development tools. The :file:`dev-requirements.txt`
+file has comments describing what we use each dependency for, so we might just
+as well show include the file verbatim here:
+
+.. literalinclude:: ../dev-requirements.txt
+
+You probably won't use all of these development tools, but at least a
+majority of them. Install them all into the active virtualenv by running `pip
+<https://pip.pypa.io/>`_::
+
+    pip install --upgrade -r dev-requirements.txt
+
+To upgrade the tools in the future, just rerun the exact same command.
+
+
+Install Mopidy from the Git repo
+--------------------------------
+
+Next up, we'll want to run Mopidy from the Git repo. There's two reasons for
+this: First of all, it lets you easily change the source code, restart Mopidy,
+and see the change take effect. Second, it's a convenient way to keep at the
+bleeding edge, testing the latest developments in Mopidy itself or test some
+extension against the latest Mopidy changes.
+
+Assuming you're still inside the Git repo, use pip to install Mopidy from the
+Git repo in an "editable" form::
+
+    pip install --editable .
```

Comment:
At this point, we're inside the virtualenv that was created with `--python $(which python2.7)`, so we should be safe.